sdk/eventhub/azure-eventhub/samples/async_samples/recv_with_checkpoint_by_batch_async.py

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env python
+
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+"""
+An example to show receiving events from an Event Hub with checkpoint store doing checkpoint by batch asynchronously.
+In the `receive` method of `EventHubConsumerClient`:
+If no partition id is specified, the checkpoint_store are used for load-balance and checkpoint.
+If partition id is specified, the checkpoint_store can only be used for checkpoint.
+"""
+
+import asyncio
+import os
+from collections import defaultdict
+from azure.eventhub.aio import EventHubConsumerClient
+from azure.eventhub.extensions.checkpointstoreblobaio import BlobCheckpointStore
+
+CONNECTION_STR = os.environ["EVENT_HUB_CONN_STR"]
+STORAGE_CONNECTION_STR = os.environ["AZURE_STORAGE_CONN_STR"]
+
+partition_recv_cnt_dict = defaultdict(int)
+checkpoint_batch_event_cnt = 20
+
+
+async def on_event(partition_context, event):
+    # put your code here
+    p_id = partition_context.partition_id
+    print("Received event from partition: {}".format(p_id))
+    partition_recv_cnt_dict[p_id] += 1
+    if partition_recv_cnt_dict[p_id] >= checkpoint_batch_event_cnt:
+        await partition_context.update_checkpoint(event)
+        partition_recv_cnt_dict[p_id] -= checkpoint_batch_event_cnt
```

Comment:
suggest change partition_recv_cnt_dict to a better name. It looks like the total receiving counts of a partition

Reply:
Another way to do is to use partition_context

```python
async def on_event(partition_context, event):
    # put your code here
    checkpoint_count = getattr(partition_context, "checkpoint_count", 0)
    checkpoint_count += 1
    if checkpoint_count >= checkpoint_batch_event_cnt:
        await partition_context.update_checkpoint(event)
        partition_count.checkpoint_count = 0
```

Reply:
I'd like to stick to the dict  implementation :P, dict name updated